app/src/main/java/com/example/mobile_development/MainActivity.kt

```diff
@@ -0,0 +1,244 @@
+        if(email != "") {
+            var isValid = email.length > 3 && EMAIL_ADDRESS_PATTERN.matcher(email).matches()
+            setIsError(!isValid)
+        return isValid
+        }
+        return  true
+    }
+
+
+    Image(
+        modifier = Modifier.fillMaxSize(),
+        painter = painterResource(R.drawable.etherealartefacts_login_background),
+        contentDescription = "background_image",
+        contentScale = ContentScale.FillBounds
+    )
+    Column(modifier = modifier.fillMaxSize()) {
+
+        Row(
+            modifier = Modifier
+                .size(width = 800.dp, height = 230.dp)
+                .padding(50.dp)
+        ) {
+            Image(
+                modifier = Modifier
+                    .size(width = 500.dp, height = 800.dp)
+                    .padding(0.dp),
+                imageVector = ImageVector.vectorResource(id = R.drawable.ethereal_artefacts_logo),
+                contentDescription = "background_image",
+                contentScale = ContentScale.Inside
+            )
+        }
+
+        Row(
+            modifier = Modifier
+                .size(width = 400.dp, height = 70.dp)
+                .padding(10.dp)
+        ) {
+            Text(
+                text = "Log in",
+                modifier = modifier,
+                fontSize = 30.sp,
+                fontWeight = FontWeight.Bold,
+            )
+        }
+
+        Row(
+            modifier = Modifier
+                .size(width = 400.dp, 100.dp)
+                .padding(10.dp)
+        ) {
+            OutlinedTextField(
+                modifier = modifier
+                    .size(width = 400.dp, height = 50.dp)
+                    .fillMaxSize(),
+                value = email,
+                onValueChange = { newValue -> setEmail(newValue) },
+                placeholder = { Text("Email") },
+                isError = !ValidateEmail(),
+                supportingText = {
+                    if (!ValidateEmail()) {
+                        Text(
+                            modifier = Modifier.fillMaxWidth(),
+                            text = "Invalid email",
+                            color = MaterialTheme.colorScheme.error
+                        )
+                    }
+                },
+            )
+        }
+
+        Row(
+            modifier = Modifier
+                .padding(10.dp)
+        ) {
+            OutlinedTextField(
+                modifier = modifier
+                    .size(width = 400.dp, height = 50.dp)
+                    .fillMaxSize(),
+                value = password,
+                onValueChange = { newValue -> setPassword(newValue) },
+                placeholder = { Text("Password") },
+                visualTransformation = if (showPass) VisualTransformation.None else PasswordVisualTransformation(),
+                trailingIcon = {
+                    val (icon, iconColor) = if (showPass) {
+                        Pair(
```

Comment:
Note: This is a weird approach. It would be an interesting implementation if it really works. Generally you would just supply an `Icon` with a tint instead of `Pair`.

Reply:
As far as I can see it works fine, if there are problems in the future I'll change it.